create.ps1

```diff
@@ -1,200 +1,220 @@
+                
+                #$sqlParameters = $account.psobject.properties | ForEach-Object -begin { $h = @{} } -process { $h."$($_.Name)" = $_.Value } -end { $h }
+
+                if (-Not($actionContext.DryRun -eq $true)) {          
+                    $null = Invoke-SqliteQuery -DataSource $database -Query $query -Verbose:$verbose
+                }
+                    
+                $outputContext.AccountReference = $account.gebruikersnaam
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        Action  = "CreateAccount" # Optionally specify a different action for this audit log
+                        Message = "Created account with username $($account.UserName)"
+                        IsError = $false
+                    })
+
+                ## Also fill roles table for each contract incondition
+
+                $contracts = $personContext.Person.Contracts
+
+                [array]$desiredContracts = $contracts | Where-Object { $_.Context.InConditions -eq $true -or $actionContext.DryRun -eq $true }
+
+                if ($desiredContracts.length -lt 1) {
+                    # no contracts in scope found
+                    throw 'No Contracts in scope [InConditions] found!'
+                }
+                elseif ($desiredContracts.length -ge 1) {
+                    # one or more contracts found
+                    foreach ($contract in $desiredContracts){
+                        
+                        $query = "INSERT OR REPLACE INTO Roles ('gebruikersnaam', 'ou', 'functie', 'oucode', 'functiecode', 'createtime') VALUES ('$($account.gebruikersnaam)','$($contract.department.displayname)','$($contract.title.name)','$($contract.department.externalid)','$($contract.title.code)',datetime());"
+                        if (-Not($actionContext.DryRun -eq $true)) {          
+                            $null = Invoke-SqliteQuery -DataSource $database -Query $query -Verbose:$verbose
+                        } else {
+                            Write-verbose -verbose "Would execute: $query"
+                        }
+                    }
+                }                    
+            }
+            catch {             
+                write-error "$($_)"
+                $ex = $PSItem
+                          
+                $auditMessage = "Error adding data to SQL Lite DB. Error: $($ex.Exception.Message)"
+                Write-Verbose "Error at Line '$($ex.InvocationInfo.ScriptLineNumber)': $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        Action  = "CreateAccount" # Optionally specify a different action for this audit log
+                        Message = $auditMessage
+                        IsError = $false
+                    })
+                throw "Error adding data to SQL Lite DB"
+                    
+            }
```

Comment:
Maybe remove the extra nested try/catch to improve error handling. 

Currently, if there's an error, we generate a detailed audit log and trigger an exception within the catch block at line 169. However, this results in entering the catch block at line 198 due to the exception, where another detailed audit log is generated and another exception is triggered. Consequently, we end up with two sets of detailed audit and verbose logs.

Furthermore, the exception thrown at line 181 causes the loss of the original error context when we reach the catch block at line 198.